components/dialog/dialog-mixin.js

```diff
@@ -238,6 +244,7 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		// handle "dialog-action" for backwards-compatibility
 		if (!e.target.hasAttribute('data-dialog-action') && !e.target.hasAttribute('dialog-action')) return;
 		const action = e.target.getAttribute('data-dialog-action') || e.target.getAttribute('dialog-action');
+		this._interceptDialogClosing = e.target.hasAttribute('data-intercept-dialog-closing');
```

Comment:
What's the reason for using this `data-` attribute + flag vs. just always firing the `d2l-dialog-abort-close` event when aborting (which isn't composed and doesn't bubble anyway) and then bypassing the abort if the event has `defaultPrevented` (like you're already doing).

Just seems like an extra unnecessary thing folks have to do to opt into this.

Reply:
Since the dialog can be closed not only by 'X' and 'Esc' buttons (they assign 'abort' action)  I suggest to introduce 'data-intercept-dialog-closing' attribute for other buttons that are able to trigger dialog closing.

Reply:
After chatting with @dbatiste, here's a suggestion:

What if instead we called the event something like `dialog-dialog-before-close` and ALWAYS fired it whenever the browser is about to close. So then you don't need the `data-` attribute, and you don't need `this._interceptDialogClosing` and I don't think you need `this._action = action ?? abortAction`.

The event would always fire, and consumers could call `preventDefault()` on it if they wanted to prevent the actual close from occurring, just like you have now. The special `force-close` action shouldn't be required either since the consumer could just _not_ call `preventDefault()` if that's what they wanted.

Reply:
Updated according to your suggestion.